build/cmake/modules/cotire.cmake

```diff
@@ -2371,7 +2371,7 @@ function (cotire_generate_target_script _language _configurations _target _targe
 		# use file(GENERATE ...) to expand generator expressions in the target script at CMake generate-time
 		set (_configNameOrNoneGeneratorExpression "$<$<CONFIG:>:None>$<$<NOT:$<CONFIG:>>:$<CONFIGURATION>>")
 		set (_targetCotireConfigScript "${CMAKE_CURRENT_BINARY_DIR}/${_target}_${_language}_${_configNameOrNoneGeneratorExpression}_${_moduleName}")
-		file (GENERATE OUTPUT "${_targetCotireConfigScript}" INPUT "${_targetCotireScript}")
+		file (GENERATE OUTPUT "${_targetCotireConfigScript}" INPUT "${_targetCotireScript}" TARGET "${_target}")
```

Comment:
Reminder: The `TARGET` keyword was added in CMake 3.19. So in the wild, this can only be used together with a test for the `CMAKE_VERSION`, unfortunately.